core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/modules/NestedJarResourceLoader.java

```diff
@@ -76,6 +76,11 @@ public static ResourceLoader loaderFor(URL base, String rootPath, String loaderP
                 }
             }
         } else if (urlString.startsWith("file:")) {
+            if (loaderName.endsWith(".jar") || loaderName.endsWith(".war")) {
+                return ResourceLoaders.createJarResourceLoader(
+                        loaderName,
+                        new JarFile(new File(new File(urlString.substring(5, urlString.length())), loaderPath)));
```

Comment:
Not a blocker, but the nested `File` object is not needed, as you can use the `File(String, String)` constructor